File: NodeJS/qsort.js
function qsort(...arr) {
  let key = arr.shift();
  return key === undefined ? [] : qsort(...arr.filter(_ => _ <= key)).concat([key]).concat(qsort(...arr.filter(_ => _ > key)));
}

console.log(qsort(1, 3, 2, 4, 6, 10, 8, 9, 11, 99, 100, 7));

File: Haskell/qsort.hs
qsort :: Ord a => [a] -> [a]
 
qsort [] = []
qsort (x:xs) = qsort l ++ [x] ++ qsort g
                 where l = filter (<x) xs
                       g = filter (>=x) xs

main = print $ qsort [1,10,2,9,3,7,2]
